### PR TITLE
Make `ockam space show` (no args) interactive

### LINKED ISSUE #6472: Make `ockam space show` (no args) interactive by asking the user to choose from a list of space names to show (tuify)
# Current behavior

When a user runs `ockam space show` without any arguments, currently they are shown some help text. Here's a screenshot of this.

Image

# Desired behavior

Change the behavior of this command so that it becomes interactive with the user when no arguments are passed. Note that this can only be allowed to happen if:

- the terminal is interactive, and,
- the [`--no-input`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L265) or `--quiet` flags have *not* been passed, and,
- no arguments have been passed in to the command.

This what the user flow should look like in interactive mode.

1) Use the [tuify](https://crates.io/crates/r3bl_tuify) crate's `select_from_list()` function to ask the user to select from a list of space names. They should be able to make multiple selections. If there are none, then simply display a message saying that  none exist (not an error).

2) Once the user has made their selection(s), then run `show` for each selected space, and display the results of the `show` command for each space that they've selected.
   - They should be able to make a single selection.
   - If they have made no selections then exit by displaying a message to the user letting them know they did not make any selection.

> The following is a video showing a generic user flow that shows the user experience when a user is asked to choose a from a list of "things". Then an action is performed on each "thing". Finally status information is reported for each action that is performed on each "thing". Please note that this flow is *not* specific to this issue. 
> Image
>  https://asciinema.org/a/612547

# Implementation details

- Please use the functions `select_multiple`, and `confirm_interactively` in this [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L281).

- Here's a similar PR to get some ideas from: https://github.com/build-trust/ockam/pull/6480

- You can learn more about how to use the `select_from_list()` in the tuify crate [here](https://docs.rs/r3bl_tuify/latest/r3bl_tuify/). Here is an [example](https://github.com/r3bl-org/r3bl_rs_utils/blob/ockam-demo/tuify/examples/main_interactive.rs) that is very similar to the steps you might have to take for this issue.
- Here's a [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L255) you can look at to detect when a terminal is interactive or not.

- Here's a [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/space/show.rs#L43) you can look at to see where the `ockam space show` command is implemented.

---
<i>

We love helping new contributors! ❤️
If you have questions or need help as you explore, please join us on [Discord](https://discord.gg/bewvnm6zqS). If you're looking for other issues to contribute to, please checkout our [good first issues](https://github.com/build-trust/ockam/labels/good%20first%20issue).
</i>

### PR DESCRIPTION
**Pull request by niss36**
## Current behavior

Running `ockam space show` without arguments just prints out some help text.

## Proposed changes

Interactively prompt the user to select which spaces they would like to show.

https://github.com/build-trust/ockam/assets/13674366/17f242d6-7ce4-4cd3-b824-d450b1ac8d4f

Resolves #6472

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/space/show.rs [hunk 1]
```diff
@@ -42,20 +41,91 @@ async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, ShowCommand)) -> mie
...
+        ShowMode::Single(name)
+    } else if opts.terminal.can_ask_for_user_input() {
+        let space_names = opts.state.spaces.list_items_names()?;
+        ShowMode::Selected(
+            opts.terminal
+                .select_multiple("Select one or more spaces to show".to_string(), space_names),
+        )
+    } else {
+        ShowMode::Default
+    };
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/space/show.rs:61]: The use cases are a little bit more nuanced than this. 

In this case, this command can only be used if the user is enrolled. You should check that with:
```rust
let identity = opts
    .state
    .identities
    .get_or_default(cmd.cloud_opts.identity.as_deref())?;
let identity_name = identity.name();
let is_enrolled = identity.is_enrolled();
```
and let the user know (e.g "The identity 'identity_name' is not enrolled. Consider running 'ockam enroll' before using this command."

Another case that is missing is when there are no spaces.

I'd recommend you to check this PR https://github.com/build-trust/ockam/pull/6665, which has been reviewed and has a more explicit way of checking the different cases. Hope it helps!

**Review by adrianbenavides — COMMENTED** [review]
Hi @niss36, thanks for your PR! It looks good, but I think we can simplify it a bit if we follow the structure of a similar PR that got reviewed/approved recently. Let me know if you have any questions 👍

**Comment by mrinalwadhwa** [comment]
@niss36  while we still have to review this change. I'm adding the `HACKTOBERFEST-ACCEPTED` label so the contribution is counted by hacktoberfest bots before the end of October. Thank you for taking time to contribute 🎉 

Please note @adrianbenavides's feedback above.

**Comment by adrianbenavides** [comment]
Hi @niss36 ! Your PR will be landed through https://github.com/build-trust/ockam/pull/6890

Thank you so much for your help!

**Pull request closed without merging**